### PR TITLE
Re-organize folders and files

### PR DESCRIPTION
**Pull request by charleskawczynski**
# Description

Since we focusing on documentation this and next week, it seems like a good idea to re-organize/group some parts of the code. This PR:

1) moves the following folders into a `Numerics/` folder:
 - `LinearSolvers/`
 - `ODESolvers/`
 - `Mesh/`
 - `DGmethods/` and `DGmethods_old`

And 2) moves `SurfaceFluxes/` into `Common/` as this will be used by `Atmos/` and `Ocean/`.

Does anyone have strong opinions about this? And are there any other things that should move around?

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -14,21 +14,25 @@ GENERATED_BL_EXAMPLES = filter!(x -> isfile(x), GENERATED_BL_EXAMPLES)
...
+    "Common" => Any[
+        "MoistThermodynamics" => "Common/MoistThermodynamics.md",
+        "SurfaceFluxes" => "Common/SurfaceFluxes.md",
+    ],
     "Utilites" => Any[],
     "Atmos" => Any[
-        "Atmos/SurfaceFluxes.md",
         "Atmos/EDMFEquations.md",
         "Microphysics" => "Atmos/Microphysics.md",
         "AtmosModel" => "Atmos/Model/turbulence.md",
```
Comment (akshaysridhar) [inline docs/make.jl:25]:
```suggestion
        "Turbulence" => "Atmos/Model/turbulence.md",
```

File: docs/make.jl [hunk 2]
```diff
@@ -14,21 +14,25 @@ GENERATED_BL_EXAMPLES = filter!(x -> isfile(x), GENERATED_BL_EXAMPLES)
...
         "AtmosModel" => "Atmos/Model/tracers.md",
```
Comment (akshaysridhar) [inline docs/make.jl:26]:
```suggestion
        "Tracers" => "Atmos/Model/tracers.md",
```

**Comment by blallen** [comment]
Love the idea, but I am concerned about the conflicts this might create with outstanding PRs

**Comment by charleskawczynski** [comment]
> Love the idea, but I am concerned about the conflicts this might create with outstanding PRs

Any PR in particular? I don't mind re-applying these changes after waiting for something else to pass through. I'll assume this folder structure in the docs until it's a good time to re-apply these changes.

**Comment by blallen** [comment]
> > Love the idea, but I am concerned about the conflicts this might create with outstanding PRs
> 
> Any PR in particular? I don't mind re-applying these changes after waiting for something else to pass through. I'll assume this folder structure in the docs until it's a good time to re-apply these changes.

#961 in particular :) Moving the DGmethods folder is going to make updating that a nightmare

**Comment by kpamnany** [comment]
IMO: Arrays and Mesh are not Numerics but distributed computing infrastructure.

**Comment by blallen** [comment]
> IMO: Arrays and Mesh are not Numerics but distributed computing infrastructure.

Agree about the Arrays, but there is a DG intrinsic component to Grids. A mesh (the map of elements in physical space) can be thought about as distributed computing infrastructure, but the details of the Gauss Lobatto points are numerics.

**Comment by jkozdon** [comment]
> #961 and #896 in particular :) Moving the DGmethods and LinearSolvers folders is going to make updating those two a nightmare

git will do fine with files that are moved around. It can track the changes through file movement (the new files in #896 will need be moved and the paths fixed). But it won't be a nightmare, git is pretty good with this sort of stuff.

**Comment by tapios** [comment]
It does seem like a good idea to move `Arrays` up a level. But otherwise LGTM.

**Comment by charleskawczynski** [comment]
> IMO: Arrays and Mesh are not Numerics but distributed computing infrastructure.

I agree with @blallen and Tapio here, Meshes are really tied to the spatial discretization. Arrays are orthogonal to time and spatial discretizations. I'll move Arrays up a level.

**Pull request merged**